src/celpy/evaluation.py

```diff
@@ -39,6 +39,7 @@
 import logging
 import operator
 import re
+import re2
```

Comment:
Seeing mypy errors. I think this needs a `# type: ignore [import-untyped]` to ignore the lack of type hints.

This module may be a little too immature if it lacks type hints. Is there another choice?

```
py38: commands[6]> poetry run mypy --show-error-codes src
src/celpy/evaluation.py:42: error: Skipping analyzing "re2": module is installed, but missing library stubs or py.typed marker  [import-untyped]
src/celpy/evaluation.py:42: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
Found 1 error in 1 file (checked 9 source files)
```

Reply:
I'm not an expert on re2 libraries, but in the interest of showing my work, here's what I found about the top results when searching for "re2 python":

- The proposed re2 library in this PR: https://github.com/google/re2 `pip install google-re2` (last update: 6 hours ago) https://snyk.io/advisor/python/google-re2 (Snyk health score: 96/100)
- https://github.com/facebook/pyre2/ `pip install pyre2` (last update: 5 years ago) https://snyk.io/advisor/python/pyre2 (Snyk health score: 57/100)
- https://github.com/axiak/pyre2/ `pip install re2` (last update: 8 years ago) https://snyk.io/advisor/python/re2 (Snyk health score: 45/100)
- https://github.com/freepn/py-re2 `pip install py-re2` (last update: 3 years ago) https://snyk.io/advisor/python/py-re2 (Snyk health score: 33/100)

I don't know how much to weigh the Snyk scores but they do give another data point in addition to the most recent commit date.

My thinking is that `google-re2` is the best of the bunch.

Regarding mypy, the various re2 packages are intended to be drop-in replacements for `re` so they should have compatible interfaces. I don't know how to tell mypy to use the same typestubs as `re` for `re2`. If you happen to know if there is an easy way to do that, please let me know.

Otherwise, I can add support in this repo for typestubs just for the portion of `google-re2` that we use, using https://mypy.readthedocs.io/en/stable/stubgen.html. 

Thoughts?

Reply:
Solid research. One that has type hints is the bare minimum for me. After that, you've ranked them nicely.